src/table.js

```diff
@@ -538,15 +543,105 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  const stringValue = typeof value === "string" ? value.trim() : value;
+  switch (type) {
+    case "string":
+      return typeof value === "string"
+        ? stringValue
+        : value || value === 0
+        ? value.toString()
+        : null;
+    case "boolean":
+      return value === true || stringValue === "true"
+        ? true
+        : value === false || stringValue === "false"
+        ? false
+        : null;
+    case "integer":
+      return value === 0
+        ? value
+        : !value || isNaN(parseInt(value))
+        ? NaN
+        : parseInt(value);
```

Comment:
I’d prefer to use `Number(value)` to coerce to a number rather than `parseInt` for two reasons. First, `parseInt` is looser: it ignores trailing non-whitespace characters, for example `parseInt(" 1 whatever ")` returns `1`, but `Number(" 1 whatever ")` returns `NaN`. Secondly, `parseInt` is going to be a lot slower if the input is _already_ numbers (or dates), since it’ll first coerce the value to a string and then parse it; it’s faster to coerce directly to a number.

I’m also not sure what our policy is on nullish values; should those be preserved or coerced to NaN? In JavaScript, `null` is coerced to `0`, but I could see us wanting to preserve nulls or coerce them to NaN. But in a case like the boolean `false` as input, I expect `0` rather than `NaN`.

I also wonder whether we need integer-specific coercion, or if `"integer"` should here be treated as an alias for `"number"` and do exactly the same thing. In other words, I don’t think it’s necessary for us to provide `"integer"` as a target type for coercion; it’s sufficient to provide `"number"`. (If you want to round your numbers to integers, you can do that yourself in JavaScript or SQL etc.)

Reply:
I think if we offer `"integer"` as a target type for coercion, it would feel a bit broken if it acted as an alias for `"number"` and returned numbers with decimals. But I'm not opposed to removing it as a target type for coercion, and only offering `"number"`.

Reply:
Yeah, my preference is to remove `"integer"` as a coercion target type too.

Reply:
@mbostock: Thank you! I believe I've addressed the issues you raised. `"integer"`, `"array"`, `"object"`, `"buffer"`, and `"other"` still appear in the `switch` statement despite being removed as target types for coercion because they can still be inferred, so we need to handle them without throwing an error. 

I moved the promotion of primitive arrays to object arrays up into `loadTableDataSource`, and have not seen any issues, but if you have any thoughts on that, I'd love to hear.